auth/credentials/internal/externalaccount/externalaccount.go

```diff
@@ -204,11 +204,24 @@ func NewTokenProvider(opts *Options) (auth.TokenProvider, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	client := opts.Client
```

Comment:
I think, at least sometimes, this needs to win over xp.client(). The way the code is today it is not possible to tell if the client in opts is user specified or the default. If it is the default client it is fine to override it, but if the user explicitly passed a client in it feels like we should honor it no matter what. Thoughts?

Reply:
I did think about that, seems to me like our options are:

1: Refactor so we are passing some signal down from the detect call that the default client is being used
2: Do a check to see if the client used is the same as the default (via reflect.deepequals) this seems hacky, but would theoretically work
3: The implementation here where we always override. I could add some log message to make it clear what's happening if that's better. 

I chose 3 here since I didn't actually see a use case where a user would want to use a different client. If they are passing in a client with TLS already set up, it defeats the purpose of using the X509 credential type, since the credentials doesn't actually use a 3rd party subject token, all of the work is in the cert/client creation logic itself

Reply:
Modified to track whether the default client is being used, and only override if that is true

Reply:
lgtm